clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1030,6 +1030,7 @@ def err_capture_default_first : Error<
 def ext_decl_attrs_on_lambda : ExtWarn<
   "%select{an attribute specifier sequence|%0}1 in this position "
   "is a C++23 extension">, InGroup<CXX23>;
+def ext_lambda : ExtWarn<"lambdas are a C++11 extension">, InGroup<CXX11>;
```

Comment:
This is not tested.  Also you probably want to move that next to warn_cxx98_compat_lambda